test/gtest/graphapi_execution_plan.cpp

```diff
@@ -0,0 +1,114 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/graphapi/execution_plan.hpp>
+
+#include <gtest/gtest.h>
+
+#include "graphapi_gtest_common.hpp"
+
+namespace {
+
+using miopen::graphapi::EngineCfg;
+using miopen::graphapi::ExecutionPlanBuilder;
+
+} // namespace
+
+TEST(GraphApi, ExecutionPlanBuilder)
+{
+    miopenHandle_t handle;
+    auto status = miopenCreate(&handle);
+    ASSERT_EQ(status, miopenStatusSuccess) << "miopenCreate() failed";
+
+    EngineCfg engineCfg;
+
+    EXPECT_NO_THROW({ ExecutionPlanBuilder().setHandle(handle).setEngineCfg(engineCfg).build(); })
+        << "ExecutionPlanBuilder failed on valid attributes";
+
+    EXPECT_NO_THROW({
+        ExecutionPlanBuilder()
+            .setHandle(handle)
+            .setEngineCfg(engineCfg)
+            .setIntermediateIds({1, 2, 3})
+            .build();
+    }) << "ExecutionPlanBuilder failed on valid attributes";
+
+    EXPECT_ANY_THROW({ ExecutionPlanBuilder().setEngineCfg(engineCfg).build(); })
+        << "ExecutionPlanBuilder failed on missing setHandle() call";
+
+    EXPECT_ANY_THROW({ ExecutionPlanBuilder().setHandle(handle).build(); })
+        << "ExecutionPlanBuilder failed on missing setEngineCfg() call";
+}
+
+namespace {
+
+class MockBackendEngineCfgDescriptor : public miopen::graphapi::BackendEngineCfgDescriptor
+{
+public:
+    MockBackendEngineCfgDescriptor() { mFinalized = true; }
+};
+
+using miopen::graphapi::GTestDescriptorAttribute;
+using miopen::graphapi::GTestDescriptorSingleValueAttribute;
+using miopen::graphapi::GTestGraphApiExecute;
+
+} // namespace
+
+TEST(GraphApi, ExecutionPlan)
+{
+    miopenHandle_t handle;
+    auto status = miopenCreate(&handle);
+    ASSERT_EQ(status, miopenStatusSuccess) << "miopenCreate() failed";
+
+    MockBackendEngineCfgDescriptor engineCfgDescriptor;
+
+    GTestDescriptorSingleValueAttribute<miopenHandle_t, char> attrHandle(
+        true,
+        "MIOPEN_ATTR_EXECUTION_PLAN_HANDLE",
+        MIOPEN_ATTR_EXECUTION_PLAN_HANDLE,
+        MIOPEN_TYPE_HANDLE,
+        MIOPEN_TYPE_CHAR,
+        2,
+        handle);
+
+    GTestDescriptorSingleValueAttribute<miopenBackendDescriptor_t, char> attrEngineCfg(
+        true,
+        "MIOPEN_ATTR_EXECUTION_PLAN_ENGINE_CONFIG",
+        MIOPEN_ATTR_EXECUTION_PLAN_ENGINE_CONFIG,
+        MIOPEN_TYPE_BACKEND_DESCRIPTOR,
+        MIOPEN_TYPE_CHAR,
+        2,
+        &engineCfgDescriptor);
+
+    GTestGraphApiExecute<GTestDescriptorAttribute*> execute;
+
+    execute.descriptor.attrsValid = true;
+    execute.descriptor.textName   = "MIOPEN_BACKEND_EXECUTION_PLAN_DESCRIPTOR";
+    execute.descriptor.type       = MIOPEN_BACKEND_EXECUTION_PLAN_DESCRIPTOR;
+    execute.descriptor.attributes = {&attrHandle, &attrEngineCfg};
+
+    execute();
```

Comment:
We may want to devise a way to test changing engineCfgDescriptor. I'm fine with leaving this for future work though.

Reply:
At the time no of our engines support configuring. After we have one, we will write a test for different configs.